src/Bridge/Doctrine/DoctrineDataStore.php

```diff
@@ -43,8 +44,9 @@ public function getPaginated(string $type, int $page, FilterCollectionInterface
         $filters->filter($qb);
 
         $pager = new Pagerfanta(new DoctrineORMAdapter($qb));
-        $pager->setCurrentPage($page);
-        $pager->setMaxPerPage($maxPerPage);
+        // If pages request is specified
+        $pager->setCurrentPage($pagesRequest ? $pagesRequest->getPage() : $page);
+        $pager->setMaxPerPage($pagesRequest ? $pagesRequest->getMaxPerPage() :$maxPerPage);
```

Comment:
The datastore should not handle anything out of its scope (generating a pager will be its limit). Please replace this code inside the controller.